invenio_notifications/manager.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2023 Graz University of Technology.
+#
+# Invenio-Notifications is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Initialization used for notifications."""
+
+
+from flask import current_app
+
+from .errors import (
+    NotificationBackendAlreadyRegisteredError,
+    NotificationBackendNotFoundError,
+)
+
+
+class NotificationManager:
+    """Notification manager taking care of backends sending notifications."""
+
+    def __init__(self, config):
+        """Constructor."""
+        self._config = config
+        self._backends = {}
+
+    def _backend_exists(self, backend_id):
+        """Check if backend is registered."""
+        return backend_id in self._backends
+
+    def _dispatch_notification(self, notification, backend, **kwargs):
+        """Extend notification and start task for sending."""
+        try:
+            extended_notification = backend.extend_notification(
+                notification.copy(), **kwargs
+            )
+
+            backend.send_notification.apply_async(
+                args=[extended_notification.dumps()],
+            )
+
+        except Exception as e:
+            current_app.logger.warning(e)
+            raise e
+
+    def broadcast(self, notification, **kwargs):
+        """Broadcast notification to backends."""
+        for backend in self._backends.values():
+            self._dispatch_notification(notification, backend, **kwargs)
```

Comment:
since in line 45 the an exception is raise again, if one backend fails the subsequent ones won't run?

Reply:
You are right, thanks!

Broadcast can probably be removed in general, as the recipient of the notfication payload is rather specific and not a user dump as first envisioned. With the `NotificationOp` it should also be easy to target different backends.